### PR TITLE
Add getDayOfWeek from epoch date

### PR DESCRIPTION
**Pull request by cooldome**
Split getDayOfWeek into two procs, capability to compute day of the week from Epoch Date is useful on its own, plus it proves to be useful for efficiency within the library

### CODE REVIEW

File: lib/pure/times.nim [hunk 1]
```diff
@@ -219,17 +219,20 @@ proc getDayOfYear*(monthday: MonthdayRange, month: Month, year: int): YeardayRan
...
+  let days = epochday - 3
   let weeks = (if days >= 0: days else: days - 6) div 7
   let wd = days - weeks * 7
   # The value of d is 0 for a Sunday, 1 for a Monday, 2 for a Tuesday, etc.
   # so we must correct for the WeekDay type.
   result = if wd == 0: dSun else: WeekDay(wd - 1)
+    
+proc getDayOfWeek*(monthday: MonthdayRange, month: Month, year: int): WeekDay {.tags: [], raises: [], benign .} =
+  ## Returns the day of the week enum from day, month and year.
+  ## Equivalent with ``initDateTime(day, month, year).weekday``.
```
Comment (Araq) [inline lib/pure/times.nim]: I know it was there before but I think correct English is  "equivalent to".

Comment (cooldome) [reply lib/pure/times.nim]: yep, all done

File: lib/pure/times.nim [hunk 2]
```diff
@@ -219,17 +219,20 @@ proc getDayOfYear*(monthday: MonthdayRange, month: Month, year: int): YeardayRan
...
     result = daysUntilMonth[month] + monthday - 1
 
-proc getDayOfWeek*(monthday: MonthdayRange, month: Month, year: int): WeekDay {.tags: [], raises: [], benign .} =
-  ## Returns the day of the week enum from day, month and year.
-  ## Equivalent with ``initDateTime(day, month, year).weekday``.
-  assertValidDate monthday, month, year
-  # 1970-01-01 is a Thursday, we adjust to the previous Monday
-  let days = toEpochday(monthday, month, year) - 3
+proc getDayOfWeek*(epochday: int64): WeekDay {.tags: [], raises: [], benign .} =
+  ## Returns the day of the week enum from epoch day
```
Comment (dom96) [inline lib/pure/times.nim]: What's an "epoch day"?

**Comment by GULPF** [comment]
I don't think this proc should be exported, since its rarely needed. When the `Date` type is implemented, it can be done with `fromEpochDay(x).weekday` instead. I know the proc is needed internally anyway, but if it's exported then soon somebody will demand a `getDayOfYear(epochday)` as well, which is also rarely useful. Also, if the module ever changes so that we no longer need it internally we will still need to keep it if it's exported.

File: lib/pure/times.nim (hunk 2 above)
Comment (GULPF) [reply lib/pure/times.nim]: An epoch day is a date stored as the number of days since 1970-01-01

**Comment by cooldome** [comment]
Closed in favour of https://github.com/nim-lang/Nim/pull/7096

**Pull request closed without merging**